src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -174,7 +174,7 @@ private static <T> boolean anyFieldMatches(@Nullable Object source, Class<T> cla
 		}
 		for (Method method : source.getClass().getMethods()) {
 			if (!Modifier.isStatic(method.getModifiers()) && clazz.isAssignableFrom(method.getReturnType())
-					&& method.isAnnotationPresent(ObjectReference.class)) {
+					&& method.isAnnotationPresent(SafeToCallMethod.class)) {

```

Comment:
I think I was talking about the clearer approach too, but I still can't figure out how the predicates should be configured so that we can perform all checks we need. As about the method name, I thought adding another word would make the signature too long; I wanted to use a common name for both fields and methods, but couldn't find it. Do you think you know how it could be called better? Maybe `anyValueMatches()`?

Reply:
That could work + short JavaDoc to explain in a sentence it is matching on fields and mehtods

Reply:
Make sure to add some JavaDoc to anyValueMatches as well as to checkValue as they are complex methods that require explanation of arguments